py/desisim/survey_release.py

```diff
@@ -0,0 +1,302 @@
+"""
+desisim.survey_release
+=============
+
+Functions and methods for mimicking a DESI release footprint and object density with adequate exposure times.
+
+"""
+from __future__ import division, print_function
+from pkg_resources import resource_filename
+import os
+import desisim
+import numpy as np
+import healpy as hp
+import fitsio
+from astropy.io import fits
+from astropy.table import Table, vstack
+from desiutil.log import get_logger
+from scipy.stats import rv_discrete
+from desitarget.targetmask import desi_mask
+import desimodel.footprint
+
+
+log = get_logger()
+
+class SurveyRelease(object):
+    """ Generate a master catalog mimicking a DESI release footprint and object density with adequate exposure times.
+    To be inputted to quickquasars.
+
+    Args:
+        mastercatalog (str): path to the master catalog.
+        data_file (str): path to the data catalog.
+        qso_only (bool): if True, will only keep QSO targets.
+        seed (int): random seed for reproducibility
+        invert (bool): if True, will shuffle the random number generation by 1-random_number.
+    """
+    def __init__(self,mastercatalog,data_file=None,qso_only=True,seed=None,invert=False):
+        self.mockcatalog=Table.read(mastercatalog,hdu=1) # Assumes master catalog is in HDU1
+        log.info(f"Obtained {len(self.mockcatalog)} objects from {mastercatalog} master catalog")
+        self.mockcatalog['TARGETID']=self.mockcatalog['MOCKID']
+        self.mockcatalog['Z']=self.mockcatalog['Z_QSO_RSD']
+        self.invert=invert
+        np.random.seed(seed)
```

Comment:
Ah this is the seed. I recommend creating a member random engine `rng_engine = np.random.default_rng(seed)` and call to generate random numbers you need. `default_rng` is the recommended since numpy 1.17